Code/Source/VehicleDynamics/DriveModels/SimplifiedDriveModel.cpp

```diff
@@ -0,0 +1,132 @@
+    {
+        PidConfiguration::Reflect(context);
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<SimplifiedDriveModel, DriveModel>()
+                ->Version(1)
+                ->Field("SteeringPID", &SimplifiedDriveModel::m_steeringPid)
+                ->Field("SpeedPID", &SimplifiedDriveModel::m_speedPid);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<SimplifiedDriveModel>("Simplified Drive Model", "Configuration of a simplified vehicle dynamics drive model")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &SimplifiedDriveModel::m_steeringPid,
+                        "Steering PID",
+                        "Configuration of steering PID controller")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &SimplifiedDriveModel::m_speedPid,
+                        "Speed PID",
+                        "Configuration of speed PID controller");
+            }
+        }
+    }
+
+    void SimplifiedDriveModel::Activate(const ChassisConfiguration& vehicleChassis)
+    {
+        m_driveWheelsData = VehicleDynamics::Utilities::GetAllDriveWheelsData(vehicleChassis);
+        m_steeringData = VehicleDynamics::Utilities::GetAllSteeringEntitiesData(vehicleChassis);
+        m_speedPid.InitializePid();
+        m_steeringPid.InitializePid();
+    }
+
+    void SimplifiedDriveModel::ApplyInputState(const VehicleInputsState& inputs, uint64_t nsDt)
+    {
+        ApplySteering(inputs.m_steering.GetValue(), nsDt);
+        ApplySpeed(inputs.m_speed.GetValue(), nsDt);
+    }
+
+    // TODO - speed and steering handling is quite similar, possible to refactor?
+    void SimplifiedDriveModel::ApplySteering(float steering, uint64_t nsDt)
+    {
+        if (m_steeringData.empty())
+        {
+            AZ_Warning("ApplySteering", false, "Cannot apply steering since no steering elements are defined in the model");
+            return;
+        }
+
+        const double nsDtSec = double(nsDt) / 1e9;
+        for (const auto& steeringElementData : m_steeringData)
+        {
+            auto steeringEntity = steeringElementData.m_steeringEntity;
+            AZ::Vector3 currentSteeringElementRotation;
+            AZ::TransformBus::EventResult(currentSteeringElementRotation, steeringEntity, &AZ::TransformBus::Events::GetLocalRotation);
+            auto currentSteeringAngle = currentSteeringElementRotation.Dot(steeringElementData.m_turnAxis);
+            double pidCommand = m_steeringPid.ComputeCommand(steering - currentSteeringAngle, nsDt);
+            if (AZ::IsClose(pidCommand, 0.0)) // TODO - use the third argument with some reasonable value which means "close enough"
```

Comment:
Maybe we can move ```0.0``` out to a constexpr constant double variable in the header file so it doesnt get lost here

Reply:
Here, pidCommand will always be compared to a zero (we are looking whether there is anything to apply).
The third argument of AZ::IsClose (which is unused, default for now) is something we might be interested in changing.